docs/source/examplecleanup.rst

```diff
@@ -6,55 +6,94 @@ End-to-End Example
 
 An End-to-End example walkthrough of a tartufo scan and the process of purging the dirty evil passwords that somehow ended up in your code commits. We will use an additional tool: ``BFG`` (https://rtyley.github.io/bfg-repo-cleaner/, more on this later!). 
 
-#. Use ``tartufo`` to scan your repository and find any secrets in its history!
 
-   Use what you've learned to scan your repo!
+#. Setup poetry if you want to use the most recent build
+
+   This project uses [Poetry] to manage its dependencies and do a lot of the heavy lifting. So you'll need to clone the tartufo repo and setup poetry!
+
+
+   .. code-block:: console
+
+      git clone git@github.com:godaddy/tartufo.git
+      # use a virtual environment!
+      python3.7 -m venv .venv
+      . .venv/bin/activate
+      
+
+   Next: [Install Poetry]  
+
+   Finally: Run ``poetry install``
+
+
+#. Clone your repo!
+
+   Select and clone the repo you want to run tartufo on
 
    .. code-block:: console
 
-      # Run Tartufo on your repo:
+      # Clone your repo, variables used later:
       GITHUBPROJECT="yourproject"
-      GITHUBREPO="yourrepo.git"
+      GITHUBREPO="myrepo.git"
       GITHUBADDRESS="github.com"
-      tartufo git@${GITHUBADDRESS}:${GITHUBPROJECT}/${GITHUBREPO}
-      # this will spit out a bunch of stuff and end with something like: Results have been saved in /tmp/tmp_jm3hyne
-      TARTUFOSCANPATH="/tmp/tmp_jm3hyne"
-      # this is the number of individual bad password hits tartufo has found:
-      ls ${TARTUFOSCANPATH} | wc -l
+      git clone --mirror git@${GITHUBADDRESS}:${GITHUBPROJECT}/${GITHUBREPO}
```

Comment:
Using `--mirror` creates a "bare" repo, which is [not currently supported](https://github.com/godaddy/tartufo/issues/12).

Instead, we should encourage users to either scan with `tartufo --repo-path <path-to-repo>` or `tartufo git@${GITHUBADDRESS}:${GITHUBPROJECT}/${GITHUBREPO}`. The former will use an already cloned repository, and will explicitly fetch all refs (which I'm assuming is the intent behind using `--mirror`), unless `--branch` is specified. The latter will clone the remote repo for the user and then act the same as though `--repo-path` were specified.

In other words, an explicit clone solely for the purpose of a `tartufo` scan is never necessary. 😄 

Also keep in mind that this usage is going to change before 2.0! #39 is next up in line for implementation, and will split the scans into different "sub-commands". So the commands will be a little bit different. There will be, for example:

* `tartufo scan-local-repo <path-to-repo>`
* `tartufo scan-remote-repo <git-url>`

The exact names of the sub-commands can certainly change, but that should at least get the idea across.

Reply:
for options specified ref: https://github.secureserver.net/CTO/guidelines/blob/master/Standards-Best-Practices/CodeScan/bfg.md

Reply:
I think "not necessary" may be accurate, but it is "useful"
It's faster to process a large repo locally, it lets you re-run things on the local without pushing them into github until they are just the way you want them (used a few times in this solution), and it allows you to more effectively split up bfg runs (also provided in this solution)

Reply:
Cool, agreed; "mirror" clones have all the info needed for tartufo to work, even if they are "bare" repos under the covers.